application/models/services/SurveyUpdater/GeneralSettings.php

```diff
@@ -0,0 +1,570 @@
+                }
+                break;
+            case static::FIELD_TYPE_USE_CAPTCHA:
+                $value = $input['usecaptcha'] ?? $this->calculateUseCaptchaOption(
+                    $input['usecaptcha_surveyaccess'] ?? null,
+                    $input['usecaptcha_registration'] ?? null,
+                    $input['usecaptcha_saveandload'] ?? null
+                );
+                if (is_null($value)) {
+                    $value = $default;
+                }
+                break;
+            case 'int':
+                $value = (int) $value;
+                break;
+            case null:
+            default:
+                $value = is_null($value)
+                    ? $default
+                    : $value;
+                break;
+        }
+
+        if ($field == 'tokenlength') {
+            $value = (int) (
+                (
+                    ($value  < 5 || $value  > 36)
+                    && $value != -1
+                )
+                ? 15
+                : $value
+            );
+        }
+        if (
+            $field == 'additional_languages'
+            && is_array($value)
+        ) {
+            if (
+                (
+                    $index = array_search(
+                        $survey->language,
+                        $value
+                    )
+                ) !== false
+            ) {
+                unset($value[$index]);
+            }
+            $value = implode(
+                ' ',
+                $value
+            );
+        }
+
+        $survey->{$field} = $value;
+
+        if (
+            !in_array(
+                $field,
+                $meta['updateFields']
+            )
+        ) {
+            $meta['updateFields'][] = $field;
+        }
+
+        return $meta;
+    }
+
+    /**
+     * Filter Input
+     *
+     * @param array $surveyId
+     * @return array
+     */
+    private function filterInput($input)
+    {
+        return array_map(function ($value) {
+            return is_string($value)
+                ? trim($value)
+                : $value;
+        }, $input);
+    }
+
+    /**
+     * Dispatch plugin event new survey settings
+     *
+     * @param int $surveyId
+     * @param array $pluginSettings
+     * @return void
+     */
+    private function dispatchPluginEventNewSurveySettings($surveyId, $pluginSettings)
+    {
+        $pluginSettings = is_array($pluginSettings) && !empty($pluginSettings)
+            ? $pluginSettings
+            : [];
+        foreach ($pluginSettings as $plugin => $settings) {
+            $settingsEvent = new PluginEvent('newSurveySettings');
+            $settingsEvent->set('settings', $settings);
+            $settingsEvent->set('survey', $surveyId);
+            $this->yiiPluginManager
+                ->dispatchEvent($settingsEvent, $plugin);
+        }
+    }
+
+    /**
+     * Dispatch plugin event before survey settings save
+     *
+     * @param int $surveyId
```

Comment:
params are wrong here, weird that Psalm didn't react. :/

Reply:
Fixed.